### PR TITLE
Fixed: 'canvas difference area' is not included in mask.

### PR DESCRIPTION
**Pull request by subiron**
Fixed: 'canvas difference area' is not included in mask.
Improve mask feature by marking canvas-only difference as yellow.
Include canvas diff in differenceCounter.
Updated unit test for fix and new mask feature.

### CODE REVIEW

File: core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java [hunk 1]
```diff
@@ -48,6 +46,8 @@
 
   private static final int INVALID_PIXEL_COLOR = 125 << ALPHA_SHIFT | 255 << RED_SHIFT | 0 << GREEN_SHIFT | 0;
```
Comment (wiiitek) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: Could we have those colors in simpler format? like i.e.` #23e416` ?

Comment (wiiitek) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: Could we have somethiing like this here?:
```java
int rgba = new java.awt.Color(r,g,b,a).getRGB();
```

Comment (subiron) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: +1 
done

File: core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java [hunk 2]
```diff
@@ -79,24 +80,26 @@ public static ImageComparisonResult compare(final BufferedImage pattern, final B
             img1Pixels, img2Pixels);
 
     // Sets not covered areas as error (red)
-    fastMarkOuterAreaAsError(minWidth, minHeight, widthDifference, heightDifference,
+    differenceCounter += fastMarkOuterAreaAsError(minWidth, minHeight, widthDifference, heightDifference,
```
Comment (Skejven) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: I'm wondering - but it is probably a good idea to count outer area as errors.
Please update comment above in line `82`.

File: core/jobs/src/test/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparisonTest.java [hunk 3]
```diff
@@ -95,6 +95,38 @@ public void testCompare_different() throws Exception {
     }
   }
 
+  @Test
+  public void testCompare_different_canvas() throws Exception {
```
Comment (Skejven) [inline core/jobs/src/test/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparisonTest.java]: Please rename this test to meet the [convention](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#tests-naming-convention) e.g. to:
`compare_differentHeightScreenshots_expectHeightDifferenceMarkedWithYellow` .

File: core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java (hunk 2 above)
Comment (subiron) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: i was thinking about saving "canvas-diff" value in separate field but 
what's the reason ?
As far as I know we even don't show differenceCounter  in report ;]

File: core/jobs/src/test/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparisonTest.java (hunk 3 above)
Comment (subiron) [inline core/jobs/src/test/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparisonTest.java]: renamed

File: core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java [hunk 4]
```diff
@@ -79,22 +79,21 @@ public static ImageComparisonResult compare(final BufferedImage pattern, final B
...
-    int[] emptyAreaA = new int[widthDifference * minHeight];
+                                               int heightDifference, int resultHeight, WritableRaster writableRaster) {
+    // fil right area
+    int[] emptyAreaA = new int[widthDifference * resultHeight];
     Arrays.fill(emptyAreaA, CANVAS_DIFF_COLOR);
-    writableRaster.setDataElements(minWidth, 0, widthDifference, minHeight, emptyAreaA);
+    writableRaster.setDataElements(minWidth, 0, widthDifference, resultHeight, emptyAreaA);
 
-    // fil area [0, minHeight, minWidth, resultHeight]
+    // fil bottom area
```
Comment (wiiitek) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: fil bottom area
   ->
 fill bottom area

?

Comment (subiron) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: fixed... but,are you familiar with  "Allow edits from maintainers." feture in github?
It's not enabled by default for no reason ;)

Comment (wiiitek) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/layout/utils/ImageComparison.java]: Thanks subiron! I didn't know about this!

**Comment by Skejven** [comment]
Please update documentation.

**Pull request merged**